apps/astarte_appengine_api/test/support/helpers/device.ex

```diff
@@ -104,7 +104,33 @@ defmodule Astarte.Helpers.Device do
     Enum.any?(result, &valid_result?(&1, interface, value))
   end
 
+  def valid_result?(result, interface, value) do
+    similar?(result, value)
+  end
+
   defp similar?(nil = _result, [] = _value), do: true
+  defp similar?("" = _result, nil = _value), do: true
+  defp similar?(result, result), do: true
+  defp similar?(result, [result]), do: true
+  defp similar?([result], [result]), do: true
+  defp similar?([result], result), do: true
+
+  defp similar?(%{"" => nil} = _result, [] = _value), do: true
+
+  defp similar?([~U[1970-01-01 00:00:00.000Z]], [0]), do: true
+  defp similar?(~U[1970-01-01 00:00:00.000Z], 0), do: true
+  defp similar?([~U[1970-01-01 00:00:00.000Z]], 0), do: true
+  defp similar?(~U[1970-01-01 00:00:00.000Z], [0]), do: true
```

Comment:
```suggestion
  defp similar?(datetime, timestamp) when is_struct?(datetime, DateTime) and is_integer(timestamp), do: DateTime.to_unix(datetime) == timestamp
```
might need to double check the unit